controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -114,6 +117,23 @@ func (a *Adapter) EnsureStatusReportedInSnapshot() (controller.OperationResult,
 		a.logger.Error(err, "Failed to update pipeline status in snapshot")
 		return controller.RequeueWithError(fmt.Errorf("failed to update test status in snapshot: %w", err))
 	}
+
+	a.logger.Info(".........status of plr", "PLR.name", a.pipelineRun.Name, "PLR.status", pipelinerunStatus)
+
+	if h.HasPipelineRunFinished(a.pipelineRun) || (pipelinerunStatus.IsAIntegrationTestStatus() && pipelinerunStatus == intgteststat.IntegrationTestStatusCancelled) {
```

Comment:
`IsAIntegrationTestStatus()` why do we need this check here (or respectivelly only here)?

Reply:
I was thinking about nil-pointer exceptions, but you're right, this is not needed here because `RequeueWithError` will ensure that we don't reach at this line with a `nil` pipelinerunStatus. Removing this.

Reply:
how exactly will solve it null pointer error? it's built on type int, it can only validate if number is from defined range in enum.

`nil.IsAIntegrationTestStatus()` wouldn't even compile

Reply:
you're right, thanks!